### PR TITLE
Adding fast configurable shader

### PR DESCRIPTION
**Pull request by dbastienMS**
should be able to replace almost all uses of existing shaders.

### CODE REVIEW

File: Assets/HoloToolkit/Utilities/Shaders/FastConfigurable.shader [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+Shader "HoloToolkit/Fast Configurable"
+{
+    Properties
+    {
+		[Header(Base Texture and Color)]
+		[Indent]
+			[Toggle] _UseVertexColor("Vertex Color Enabled?", Float) = 0
+			[Toggle] _UseMainColor("Main Color Enabled?", Float) = 0
+			_Color("Main Color", Color) = (1,1,1,1)
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Shaders/FastConfigurable.shader]: Love it! Might I suggest a highlight property in here? So we can highlight a material when the user's gaze enters.
I'm open to other suggestions on that but I think we should have it :)

Comment (dbastienMS) [inline Assets/HoloToolkit/Utilities/Shaders/FastConfigurable.shader]: It said an email response would show up here but that seems untrue :P

I'm potentially OK with it - is this something that couldn't be accomplished through swapping to a material with a different diffuse color though?

Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Utilities/Shaders/FastConfigurable.shader]: Swapping materials could be bad. It's been known in the past to create mem leaks in unity if improperly implemented.

**Comment by dbastienMS** [comment]
The idea would be to swap materials.  Having 2 is no worse than having one.

What you're talking about is touching the material at runtime which creates a second.

Not really a leak but you can end up making a lot of materials and wasting memory.

On Jan 17, 2017 11:36 AM, Stephen Hodgson <notifications@github.com> wrote:

@HodgsonSDAS commented on this pull request.

________________________________

In Assets/HoloToolkit/Utilities/Shaders/FastConfigurable.shader<https://github.com/Microsoft/HoloToolkit-Unity/pull/454>:

> +
+// Textured ambient+diffuse:
+// Stats for Vertex shader:
+//        d3d11: 24 avg math (9..44)
+// Stats for Fragment shader:
+//        d3d11: 2 avg math (1..5), 0 avg texture (0..2)
+
+Shader "HoloToolkit/Fast Configurable"
+{
+    Properties
+    {
+               [Header(Base Texture and Color)]
+               [Indent]
+                       [Toggle] _UseVertexColor("Vertex Color Enabled?", Float) = 0
+                       [Toggle] _UseMainColor("Main Color Enabled?", Float) = 0
+                       _Color("Main Color", Color) = (1,1,1,1)


Swapping materials could be bad. It's been known in the past to create mem leaks in unity if improperly implemented.

-

File: Assets/HoloToolkit/Utilities/Shaders/FastConfigurable.shader (hunk 1 above)
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Shaders/FastConfigurable.shader]: Let's add highlights using the emission mask using a test scene in a following PR.

**Comment by NeerajW** [comment]
Yeah updating the README is good. Then we will merge this PR. 
Test scene etc. with more recommendations will come in a following PR.

**Comment by dbastienMS** [comment]
Updating the readme is definitely good, however since until we remove the old ones we'll have multiple tools that do the same job I'm not quite sure how to message it.

My preference would be to get this in and get to the business of removing the old stuff as quickly as possible so we can have a cleaner message.

**Pull request merged**